@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -216,13 +216,27 @@ function computePointTextOffset(
     offset.x = textElement.xOffset;
     offset.y = textElement.yOffset;
 
+    switch (placement.h) {
+        case HorizontalPlacement.Left:
+            // In practice applies: -= pointLabelMargin.x
+            offset.x += textElement.bounds!.max.x;
+            break;
+        case HorizontalPlacement.Right:
+            // Applies: += pointLabelMargin.x
+            offset.x -= textElement.bounds!.min.x;
+            break;
+    }
```

Comment:
This code should actually be there regardless of the margin applied, cause without it we assume that text bounding box starts or ends from (0,0), if TextCanvas measurement would do it differently then our offset may be incorrect, in the way that this old code does not support BB placed differently then from 0,0 point.